libs/filamat/src/eiff/ChunkContainer.cpp

```diff
@@ -18,8 +18,8 @@
 
 namespace filamat {
 
-void ChunkContainer::addChild(Chunk* chunk) {
-    mChildren.push_back(chunk);
+void ChunkContainer::addChild(ChunkPtr&& chunk) {
+    mChildren.push_back(std::move(chunk));
```

Comment:
Should we make a version of of `addChild` that takes all the parameters required to build a chunk so we could do an `emplace_back`?

Reply:
`addChild` would need to be templatized, but I think something like this should work:

```
    template <typename ChunkType, typename... Args>
    void addChild(Args&&... args) {
        mChildren.emplace_back(new ChunkType(std::forward<Args>(args)...));
    }
```